app/controllers/charges_controller.rb

```diff
@@ -0,0 +1,31 @@
+class ChargesController < ApplicationController 
+
+  def new
+    @stripe_btn_data = {
+      key: "#{ Rails.configuration.stripe[:publishable_key] }",
+      description: "PhotoChamp Upgrade - #{current_user.email}",
+      amount: Amount.default
+    }
+  end
+
+  def create
+    customer = Stripe::Customer.create(
+      email: current_user.email,
+      card: params[:stripeToken]
+    )
+
+    charge = Stripe::Charge.create(
```

Comment:
@levicole, yeah, a subscription will be a better way to fund some of the services this will need.